Formula/o/openjdk@8.rb

```diff
@@ -187,3 +192,2786 @@ class HelloWorld {
     assert_match "Hello, world!", shell_output("#{bin}/java HelloWorld")
   end
 end
+
+__END__
+Subject: [PATCH] Fixed C++11 build failure issue
```

Comment:
Can you share more about what this patch does? Did you make this patch, or does it come from somewhere else? We cannot accept such a nontrivial patch without a strong and valid reason.

Also, patches of this size should live in https://github.com/Homebrew/formula-patches.

Reply:
Yes, I made it. I refer to MacPorts' build [Patch files](https://github.com/macports/macports-ports/tree/master/java/openjdk8/files).

Are there guidelines on how to create a PR to [formula-patches](https://github.com/Homebrew/formula-patches)? How does it relate to this PR here?

Reply:
> I refer to MacPorts' build [Patch files](https://github.com/macports/macports-ports/tree/master/java/openjdk8/files).

In that case, can you reference the MacPorts patches directly instead of inlining them? (Just add the necessary ones, as I don't think all of them are required for the macOS versions we support.) See [`enscript.rb`](https://github.com/Homebrew/homebrew-core/blob/aaaef4633590c9ef6ba81d9a674573ced9751771/Formula/e/enscript.rb#L33-L36) for example.

> Are there guidelines on how to create a PR to [formula-patches](https://github.com/Homebrew/formula-patches)? How does it relate to this PR here?

If you made any other changes on top of MacPorts' patches, place them in the `openjdk@8` directory, and make a PR. I don't think there are guidelines for that, but you may refer to the [merged PRs](https://github.com/Homebrew/formula-patches/pulls?q=is%3Apr+is%3Amerged).

Are those patches endorsed upstream?

Reply:
> > I refer to MacPorts' build [Patch files](https://github.com/macports/macports-ports/tree/master/java/openjdk8/files).
> 
> In that case, can you reference the MacPorts patches directly instead of inlining them? (Just add the necessary ones, as I don't think all of them are required for the macOS versions we support.) See [`enscript.rb`](https://github.com/Homebrew/homebrew-core/blob/aaaef4633590c9ef6ba81d9a674573ced9751771/Formula/e/enscript.rb#L33-L36) for example.
> 
> > Are there guidelines on how to create a PR to [formula-patches](https://github.com/Homebrew/formula-patches)? How does it relate to this PR here?
> 
> If you made any other changes on top of MacPorts' patches, place them in the `openjdk@8` directory, and make a PR. I don't think there are guidelines for that, but you may refer to the [merged PRs](https://github.com/Homebrew/formula-patches/pulls?q=is%3Apr+is%3Amerged).
> 
> Are those patches endorsed upstream?

I got it.

However, it may be due to the different versions of the OpenJDK version(MacPorts Now is [8u372](https://ports.macports.org/port/openjdk8/summary/)), so it is impossible to directly apply patch files in Macports.

Reply:
MacPorts patches are for 10.7 compatibily but we support 10.11 minimum. Are you sure we actually need them?

Reply:
> In that case, can you reference the MacPorts patches directly instead of inlining them? (Just add the necessary ones, as I don't think all of them are required for the macOS versions we support.) See [`enscript.rb`](https://github.com/Homebrew/homebrew-core/blob/aaaef4633590c9ef6ba81d9a674573ced9751771/Formula/e/enscript.rb#L33-L36) for example.

OK, I got it.

But there are still the same problem left now (as follows). This problem is mentioned in the previous version (1.8.0-382) of this [PR](https://github.com/Homebrew/homebrew-core/pull/137300#issuecomment-1695873698), but it has not been explained how to fix it.

- On MacOS 12 & 13:
```
Error: 2 failed steps!
brew test --retry --verbose blazegraph
brew test --retry --verbose jmxtrans
Error: Process completed with exit code 1.
```
- On MacOS 14:
```
Warning: 1 failed step ignored!
brew test --retry --verbose embulk
```
> 
> Are those patches endorsed upstream?

I think so.

Reply:
> MacPorts patches are for 10.7 compatibily but we support 10.11 minimum. Are you sure we actually need them?

I have now pushed a new commit, and now I only included the smallest required patches. I have tested these files locally, which helps to make [`openjdk@8`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/o/openjdk%408.rb) can be compiled successfully. Now these Patches are actually to solve the issues caused by stricter restrictions of Clang (C++ 11) compiler, rather than the OS version limit.
 
> The `JavaNativeFoundation` changes make sense and that change alone should at least fix the error we saw in CI. The patches however aren't clear - were there further errors you saw? It could do with a note here on what precise error you saw that you are trying to fix with them. This will also make it clear when the patches are redundant.

- Patch files [`0001-8181503-Can-t-compile-hotspot-with-c-11.patch`](https://raw.githubusercontent.com/macports/macports-ports/04ad4a17332e391cd359271965d4c6dac87a7eb2/java/openjdk8/files/0001-8181503-Can-t-compile-hotspot-with-c-11.patch) and [`0006-Disable-C-11-warnings.patch`](https://raw.githubusercontent.com/macports/macports-ports/04ad4a17332e391cd359271965d4c6dac87a7eb2/java/openjdk8/files/0006-Disable-C-11-warnings.patch) solved problems such as their file names and descriptions in their comments!

- Embedded (END) patches:
  + `jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp`
```
/private/tmp/openjdkA8-20240304-44197-jia1bt/jdk8u-jdk8u402-ga/jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp:295:12: error: cannot initialize return object of type 'jobject' (aka '_jobject *') with an rvalue of type 'bool'
    return false;
           ^~~~~
1 error generated.
make[2]: *** [/private/tmp/openjdkA8-20240304-44197-jia1bt/jdk8u-jdk8u402-ga/build/release/jdk/objs/libunpack/jni.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libs-only] Error 2
make: *** [jdk-only] Error 2
```

  + `jdk/src/macosx/native/com/sun/media/sound/PLATFORM_API_MacOSX_Ports.cpp`
```
/private/tmp/openjdkA8-20240305-26444-en6juk/jdk8u-jdk8u402-ga/jdk/src/macosx/native/com/sun/media/sound/PLATFORM_API_MacOSX_Ports.cpp:612:107: error: non-constant-expression cannot be narrowed from type 'int' to 'AudioObjectPropertyElement' (aka 'unsigned int') in initializer list [-Wc++11-narrowing]
                const AudioObjectPropertyAddress address = {kAudioObjectPropertyElementName, port->scope, ch};
                                                                                                          ^~
/private/tmp/openjdkA8-20240305-26444-en6juk/jdk8u-jdk8u402-ga/jdk/src/macosx/native/com/sun/media/sound/PLATFORM_API_MacOSX_Ports.cpp:612:107: note: insert an explicit cast to silence this issue
                const AudioObjectPropertyAddress address = {kAudioObjectPropertyElementName, port->scope, ch};
                                                                                                          ^~
                                                                                                          static_cast<AudioObjectPropertyElement>( )
1 error generated.
make[2]: *** [/private/tmp/openjdkA8-20240305-26444-en6juk/jdk8u-jdk8u402-ga/build/release/jdk/objs/libjsound/PLATFORM_API_MacOSX_Ports.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [libs-only] Error 2
make: *** [jdk-only] Error 2
```